realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmExtensions.kt

```diff
@@ -26,32 +21,58 @@ inline fun <reified T : RealmModel> Realm.where(): RealmQuery<T> {
 /**
  * Deletes all objects of the specified class from the Realm.
  *
- * @param <T> the class of the object which is to be queried for.
+ * @param T the class of the object which is to be queried for.
  * @throws IllegalStateException if the corresponding Realm is closed or called from an incorrect thread.
  */
 inline fun <reified T : RealmModel> Realm.delete() {
     return this.delete(T::class.java)
 }
 
-// TODO: Figure out if we should include this is or. Using this makes it possible to do
-// ```
-// realm.callTransaction { createObject<Foo>() }
-//
-//inline fun <T> Realm.callTransaction(crossinline action: Realm.() -> T): T {
-//    val ref = AtomicReference<T>()
-//    executeTransaction {
-//        ref.set(action(it))
-//    }
-//    return ref.get()
-//}
+/**
+ * Invokes the `action` provided to this function, using a transactional Realm instance and
+ * returns the value that is returned from the action.
+ *
+ * ### Example Usage:
+ * ``` kotlin
+ * val realm = Realm.getDefaultInstance()
+ * val (person, dog) = realm.callInTransaction {
+ *
+ *     val pet = createObject<Dog>().apply {
+ *         name = "Tony"
+ *         age = 2
+ *     }
+ *     val person = createObject<Person>(101).apply {
+ *         name = "Andy Kotlin"
+ *         dog = pet
+ *     }
+ *     Pair(person, pet)  // becomes T, can be anything.
+ * }
+ *
+ * // Andy Kotlin has a 2 year old dog named Tony
+ * println("${person.name} has a ${dog.age} year old dog named ${dog.name}")
+ * ```
+ *
+ * @param action A function that operates on a receiver type Realm that is in a transaction.
+ *               The return value of `action` is returned by this function.
+ * @return T the object returned from the `action`.
+ *
+ */
+inline fun <T> Realm.callInTransaction(crossinline action: Realm.() -> T): T {
```

Comment:
Is this naming idiomatic in Kotlin? To me, it doesn't convey what is actually happening, but it might just be my inexperience.

Reply:
Agree..  Open to ideas.  I guess it's similar to `[let](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/let.html)` functionality but transactional and specific to Realm.    Or perhaps, we should just drop it altogether if you feel it's not applicable to all??

It's not a core functionality, more of a convenience function that might not be used by all.  It's a great way to get objects out of a Tx block and makes for a clean API feel IMO.  However, the use case is that you're probably using this in a BG thread (since it's blocking) and presumably you can just do all work inside the transaction anyway (and not need to pass objects back).

Reply:
I'm not inherently against it, we should just be careful about overloading our extension function library with all sorts of ideas. Mostly because we are going to need to support them going forward and we need to ensure that they are widely applicable.

At this point in time, I don't really have a feeling for how useful this is. I mean it isn't something I missed in the Java API, but others might feel differently?

Then of course there is an entire naming discussion as well. Good with the `let` reference which also is a horrible name IMO :)

Reply:
In light of this discuss, I say let's keep it out for now. Release this first extensions support and revisit later.

Reply:
Sounds good to me